packages/app/src/runs/modals/SelectCloudProjectModal.vue

```diff
@@ -285,31 +285,41 @@ const isInternalServerError = ref(false)
 const graphqlError = ref<{ extension: string, message: string} | undefined>()
 const projectName = ref(props.gql.currentProject?.title || '')
 const projectAccess = ref<'private' | 'public'>('private')
-const organizations = computed(() => {
-  return sortBy(props.gql.cloudViewer?.organizations?.nodes.map((org) => {
+const organizationOptions = computed(() => {
+  const options = props.gql.cloudViewer?.organizations?.nodes?.map((org) => {
     return {
-      ...org,
-      projects: {
-        ...org.projects,
-        nodes: sortBy(org.projects?.nodes, 'name'),
-      },
+      id: org.id,
+      name: org.name,
       icon: FolderIcon,
     }
-  }) || [], 'name')
+  })
+
+  return sortBy(options || [], 'name')
 })
-const pickedOrganization = ref(organizations.value.length >= 1 ? organizations.value[0] : undefined)
+const pickedOrganization = ref(organizationOptions.value.length >= 1 ? organizationOptions.value[0] : undefined)
+
+const projectOptions = computed(() => {
+  const organization = props.gql.cloudViewer?.organizations?.nodes?.find((org) => org.id === pickedOrganization?.value?.id)
+  const options = organization?.projects?.nodes?.map((project) => {
+    return {
+      id: project.id,
+      name: project.name,
+      slug: project.slug,
+    }
+  })
 
-const projects = computed(() => pickedOrganization.value?.projects?.nodes || [])
-const newProject = ref(projects.value.length === 0)
-const pickedProject = ref<CloudProjectNodeFragment>()
+  return sortBy(options || [], 'name')
+})
+const newProject = ref(projectOptions.value.length === 0)
+const pickedProject = ref<typeof projectOptions.value[number]>()
 
-watchEffect(() => {
-  if (pickedOrganization.value?.projects?.nodes?.length === 1) {
-    pickedProject.value = pickedOrganization.value.projects.nodes[0]
+watch([projectOptions], () => {
```

Comment:
How come we changed to this syntax? I think `watch` + `immediate: true` is the same as `watchEffect`?

Also I think this is now actually doing something a little different - if you use `watch`, you need to get the new value from the callback - [docs](https://vuejs.org/guide/essentials/watchers.html#basic-example). Eg:

```js
watch(projectOptions, (newProjectOptions) => {
                       ^^^^^^^ this is the latest value
  // stuff
})
```

Reply:
Good call. My refactor meandered through a state where I had an reference loop between two `watchEffect`s hence the change to this format, but has since changed further so I can switch back to `watchEffect`